### PR TITLE
Short-term isFixed fix for layouts

### LINKED ISSUE #3364: amp-analytics does not load sometimes
## Short description of your issue:

I will provide more information as I can get it, but I've noticed times where the analytics code will not fire at all. There aren't any network issues like the scripts not loading or anything, so I think there must be some sort of race condition within the AMP code. Maybe something with how the "visible" trigger works?

## How do we reproduce the issue?

1. Refresh on an amp-enabled page with amp-analytics installed with GA configuration
2. Look in network requests
3. Every once in a while the GA pixel will not get sent.

## What browsers are affected?

Chrome is what I've tested on

## Which AMP version is affected?

Latest

### PR DESCRIPTION
**Pull request by dvoytenko**
Closes #3364.

### CODE REVIEW

File: src/service/resources-impl.js [hunk 1]
```diff
@@ -1779,6 +1783,27 @@ export class Resource {
...
     }
     this.layoutBox_ = box;
+
+    // Calculate whether the element is currently is or in `position:fixed`.
+    let isFixed = false;
+    if (this.isDisplayed()) {
+      const win = this.resources_.win;
+      const body = win.document.body;
+      const viewport = this.resources_.viewport_;
+      for (let n = this.element; n && n != body; n = n.offsetParent) {
```
Comment (cramforce) [inline src/service/resources-impl.js]: TIL about `offsetParent`. I couldn't find where it is supported. Everywhere we care?

Comment (dvoytenko) [inline src/service/resources-impl.js]: Hmm. I always thought it was there from beginning. I'll check the support.

Comment (dvoytenko) [inline src/service/resources-impl.js]: Yep. It's been there since IE6 :)

Comment (dvoytenko) [inline src/service/resources-impl.js]: So, there are some differences in interpretation in browsers for `offsetParent`. But none that play any role in this case.

**Comment by dvoytenko** [comment]
@cramforce PTAL. Tests added.

File: src/base-element.js [hunk 2]
```diff
@@ -189,6 +189,16 @@ export class BaseElement {
   }
 
   /**
+   * Intended to be implemented by subclasses. Tests whether the element
+   * requires fixed positioning.
+   * @return {boolean}
+   * @protected
+   */
+  isFixedRequired() {
```
Comment (cramforce) [inline src/base-element.js]: Maybe rename to `isAlwaysFixed`?

File: src/service/resources-impl.js [hunk 3]
```diff
@@ -1008,7 +1008,7 @@ export class Resources {
       // Note that when the document is not visible, neither are any of its
       // elements to reduce CPU cycles.
       const shouldBeInViewport = (this.visible_ && r.isDisplayed() &&
-          r.overlaps(visibleRect));
+          (r.isFixed() || r.overlaps(visibleRect)));
```
Comment (cramforce) [inline src/service/resources-impl.js]: I'd prefer if the `isFixed` call would be in `overlaps` and `overlaps` would use a `0,0` rect in that case. That way stuff that is fixed but at `-1000px` does not get rendered.

File: src/base-element.js (hunk 2 above)
Comment (dvoytenko) [inline src/base-element.js]: Done

File: src/service/resources-impl.js (hunk 3 above)
Comment (dvoytenko) [inline src/service/resources-impl.js]: I'd prefer that too. The problem with this is, essentially, the `layoutRect` is implemented completely incorrectly for fixed elements. And comparing overlap here with `visibleRect` is also incorrect. This functionality has to be refactored in each separate layer, which is a lot bigger fix (#3434).

**Comment by dvoytenko** [comment]
@cramforce Added TODO and comments for fixed for layers

**Comment by cramforce** [comment]
LGTM

**Pull request merged**